File: src/main/java/com/alibou/book/book/BookResponse.java
package com.alibou.book.book;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class BookResponse {
    private Long id;
    private String title;
    private String authorName;
    private String isbn;
    private String synopsis;
    private String cover;
    private Boolean archived;
    private Boolean shareable;
    private Double rate;
    private String owner;  // This is the username string
    private Long ownerId;  // Added field for owner's ID
    private Genre genre;   // Added genre field
}

File: book-network-ui/src/app/services/models/book-response.ts
/* tslint:disable */
/* eslint-disable */
export interface BookResponse {
  archived?: boolean;
  authorName?: string;
  cover?: Array<string>;
  id?: number;
  isbn?: string;
  owner?: string;
  rate?: number;
  shareable?: boolean;
  synopsis?: string;
  title?: string;
  genre?: string; // Add this line
}

export interface PageResponse<T> {
  content?: T[];
  pageable?: Pageable;
  totalPages?: number;
  totalElements?: number;
  last?: boolean;
  size?: number;
  number?: number;
  sort?: Sort;
  numberOfElements?: number;
  first?: boolean;
  empty?: boolean;
}

interface Pageable {
  sort?: Sort;
  offset?: number;
  pageNumber?: number;
  pageSize?: number;
  paged?: boolean;
  unpaged?: boolean;
}

interface Sort {
  empty?: boolean;
  sorted?: boolean;
  unsorted?: boolean;
}
